### PR TITLE
🚀 Google Classroom & Google Docs

### PR DESCRIPTION
**Pull request by EncryptedDev**
### last and final time kew
# Google Classroom
## Change log
- Adds localisation support;
- Adds a privacy mode which can be toggled in the presence settings;
- Cleans up unless code 

## Images
*All proof has been taken in English, but works for other language. Tested in Dutch and it works too*
Page | Image
------------ | -------------
Presence Settings | 
Calendar | 
Classwork - privacy mode enabled | 
Class members - privacy mode enabled | 

## Extra
The privacy mode was requested by another user (#3196)

# Google Docs
## Change log
- Fixes old and outdated code;
- Adds localisation support;
- Adds privacy mode

## Images
*All proof has been taken in English, but works for other language. Tested in Dutch and it works too*
Page | Image
------------ | -------------
Presence Settings | 
Google Docs - editing with privacy disabled | 
Google Sheets - editing with privacy enabled | 
Google Slides  - browsing | 

## Extra
Can I be author as Callum hasn't done anything here nor anything on this presence before, he just 'owns' it? kekw

### CODE REVIEW

**Review by StrikerFRFX — CHANGES_REQUESTED** [review]
codacy

**Review by StrikerFRFX — APPROVED** [review]

**Comment by StrikerFRFX** [comment]
why did you request me again? lolwhat

**Review by Bas950 — CHANGES_REQUESTED** [review]
Update fork, bump versions, upload new proof.

**Comment by EncryptedDev** [comment]
> Update fork, bump versions, upload new proof.

- Bumped versions!

What new proof do you want? And what fork do you mean?

File: websites/G/Google Classroom/dist/metadata.json [hunk 1]
```diff
@@ -3,10 +3,20 @@
...
     "en": "Google Classroom is a free web service, developed by Google for schools, that aims to simplify creating, distributing, and grading assignments in a paperless way. The primary purpose of Google Classroom is to streamline the process of sharing files between teachers and students.",
-    "nl": "Google Classroom is een gratis en makkelijke tool waarmee docenten voortgang efficiënt kunnen beheren en beoordelen, en waarmee ze in contact kunnen blijven met leerlingen, zelfs onderweg of vanuit huis."
+    "nl": "Google Classroom is een gratis en makkelijke tool waarmee docenten voortgang efficiënt kunnen beheren en beoordelen, en waarmee ze in contact kunnen blijven met leerlingen, zelfs onderweg of vanuit huis.",
+    "fr": "Google Classroom est service web gratuit, développé par Google pour les écoles, qui a pour but de simplifier la création, la distribution, et la notation des devoirs de manière numérique. Le but primaire de Google Classroom est de simplifier le processus de partage de fichiers entre les professeurs et les étudiants."
+  },
+  "url": "classroom.google.com",
+  "version": "1.1.8",
   },
   "url": "classroom.google.com",
   "version": "1.1.7",
```
Comment (Bas950) [inline websites/G/Google Classroom/dist/metadata.json:22]: 2 versions 2 urls?

File: websites/G/Google Classroom/presence.ts [hunk 2]
```diff
@@ -1,22 +1,57 @@
 const presence = new Presence({
   clientId: "632293282847784973"
 });
-let presenceData: PresenceData = {
-  largeImageKey: "logo"
-};
+
+const browsingStamp = Math.floor(Date.now() / 1000);
```
Comment (Bas950) [inline websites/G/Google Classroom/presence.ts:5]: merge consts

**Review by Bas950 — CHANGES_REQUESTED** [review]

File: websites/G/Google Docs/presence.ts [hunk 3]
```diff
@@ -1,98 +1,98 @@
...
-          presenceData.details = "Browsing documents";
-        } else {
-          presenceData.largeImageKey = "docslogo";
-          presenceData.details = "Viewing a document:";
-          presenceData.state = document.title.replace("- Google Docs", "");
-        }
+        presenceData.details = privacy
+          ? "Viewing a document"
+          : "Viewing a document:";
+        presenceData.state = privacy ? null : title;
```
Comment (Bas950) [inline websites/G/Google Docs/presence.ts:28]: if !privacy then do Viewing document: you dont need "a"
and dont set state to null, just dont set it

File: websites/G/Google Docs/presence.ts [hunk 4]
```diff
@@ -1,98 +1,98 @@
...
+    title = document.title.replace("- Google Forms", "");
     if (window.location.pathname.toLowerCase().includes("/edit")) {
-      presenceData.largeImageKey = "formslogo";
-      presenceData.details = "Editing a form:";
-      presenceData.state = document.title.replace("- Google Forms", "");
-    } else {
-      presenceData.largeImageKey = "formslogo";
+      presenceData.details = privacy ? "Editing a form" : "Editing a form:";
+      presenceData.state = privacy ? null : title;
+    } else if(document.location.pathname === "/forms/u/0/"){
```
Comment (Bas950) [inline websites/G/Google Docs/presence.ts:39]: same here

File: websites/G/Google Docs/presence.ts [hunk 5]
```diff
@@ -1,98 +1,98 @@
...
       } else {
-        presenceData.largeImageKey = "formslogo";
         if (window.location.pathname.toLowerCase() === "/forms/u/0") {
           presenceData.details = "Browsing forms";
         } else {
-          presenceData.largeImageKey = "formslogo";
-          presenceData.details = "Viewing a form:";
-          presenceData.state = document.title.replace("- Google Forms", "");
+          presenceData.details = privacy ? "Viewing a form" : "Viewing a form:";
+          presenceData.state = privacy ? null : title;
```
Comment (Bas950) [inline websites/G/Google Docs/presence.ts:47]: and here

File: websites/G/Google Docs/presence.ts [hunk 6]
```diff
@@ -1,98 +1,98 @@
...
     presenceData.largeImageKey = "sheetslogo";
+    title = document.title.replace("- Google Sheets", "");
     if (window.location.pathname.toLowerCase().includes("/edit")) {
-      presenceData.largeImageKey = "sheetslogo";
-      presenceData.details = "Editing a spreadsheet:";
-      presenceData.state = document.title.replace("- Google Sheets", "");
+      presenceData.details = privacy
+        ? "Editing a spreadsheet"
+        : "Editing a spreadsheet:";
+      presenceData.state = privacy ? null : title;
```
Comment (Bas950) [inline websites/G/Google Docs/presence.ts:60]: 1more, and werent there strings for this?

File: websites/G/Google Docs/presence.ts [hunk 7]
```diff
@@ -1,98 +1,98 @@
...
         if (window.location.pathname.toLowerCase() === "/spreadsheets/u/0") {
           presenceData.details = "Browsing spreadsheets";
         } else {
-          presenceData.largeImageKey = "sheetslogo";
-          presenceData.details = "Viewing a spreadsheet:";
-          presenceData.state = document.title.replace("- Google Sheets", "");
+          presenceData.details = privacy
+            ? "Viewing a spreadsheet"
+            : "Viewing a spreadsheet:";
+          presenceData.state = privacy ? null : title;
```
Comment (Bas950) [inline websites/G/Google Docs/presence.ts:71]: again

File: websites/G/Google Docs/presence.ts [hunk 8]
```diff
@@ -1,98 +1,98 @@
...
           presenceData.details = "Browsing slidesheets";
         } else {
-          presenceData.largeImageKey = "slideslogo";
-          presenceData.details = "Viewing a slidesheet:";
-          presenceData.state = document.title.replace("- Google Slides", "");
+          presenceData.details = privacy
+            ? "Viewing a slidesheet"
+            : "Viewing a slidesheet:";
+          presenceData.state = privacy ? null : title;
         }
```
Comment (Bas950) [inline websites/G/Google Docs/presence.ts:96]: and last 2

**Review by Bas950 — CHANGES_REQUESTED** [review]

File: websites/G/Google Docs/presence.ts (hunk 3 above)
Comment (EncryptedDev) [reply websites/G/Google Docs/presence.ts:28]: bruh moment - yeah idk what happened there because it wasn't a problem before lmao

**Review by Bas950 — CHANGES_REQUESTED** [review]
Upload new proof

**Review by StrikerFRFX — CHANGES_REQUESTED** [review]
Upload new proof of working please

**Comment by EncryptedDev** [comment]
image
image
image
image
image

**Review by Bas950 — APPROVED** [review]

**Review by StrikerFRFX — APPROVED** [review]

**Pull request merged**